### PR TITLE
Update `rstudio.com` to `posit.co` in urls

### PR DESCRIPTION
**Pull request by zekiakyol**
### Intent

This pull request changes documentation at [RStudio IDE User Guide](https://docs.posit.co/ide/user/) and README by
- changing `rstudio.com` to `posit.co` in all suitable urls
- changing deprecated `https://r4ds.hadley.nz/data-visualisation.html` to `https://r4ds.hadley.nz/data-visualize`
- changing deprecated `http://r-pkgs.had.co.nz/` to `https://r-pkgs.org/`
- changing r4ds to r4ds(2e), i.e. `https://www.oreilly.com/library/view/r-for-data/9781491910382/` to `https://www.oreilly.com/library/view/r-for-data/9781492097396/`
- adding a missing `*` for bolding.

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Review by olivroy — COMMENTED** [review]
I think this is the correct link. part of #13626

**Comment by zekiakyol** [comment]
You are correct! I updated the url according to your suggestion.

File: docs/user/rstudio/ide/guide/tools/jobs.qmd [hunk 1]
```diff
@@ -3,9 +3,9 @@ title: "RStudio Jobs"
...
+Local background jobs are supported by all versions of the RStudio IDE, server and desktop. Remote background jobs are a feature of [Posit Workbench](https://posit.co/products/enterprise/workbench/) and are orchestrated by the [Launcher](https://docs.posit.co/ide/server-pro/job_launcher/job_launcher.html), which also supports running interactive sessions on remote resource managers like [Kubernetes](https://kubernetes.io/). More information about configuring the Job Launcher can be found in the [Posit Workbench Admin Guide](https://docs.posit.co/ide/server-pro/job_launcher/job_launcher.html#job-launcher), and more information about using Workbench Jobs can be found in the [Posit Workbench User Guide](https://docs.posit.co/ide/server-pro/user/rstudio-pro/guide/workbench-jobs.html).
```
Comment (jgutman) [inline docs/user/rstudio/ide/guide/tools/jobs.qmd:8]:
```suggestion
Local background jobs are supported by all versions of the RStudio IDE, server and desktop. Remote background jobs are a feature of [Posit Workbench](https://posit.co/products/enterprise/workbench/) and are orchestrated by the [Launcher](https://docs.posit.co/ide/server-pro/job_launcher/job_launcher.html), which also supports running interactive sessions on remote resource managers like [Kubernetes](https://kubernetes.io/). More information about configuring the Job Launcher can be found in the [Posit Workbench Admin Guide](https://docs.posit.co/ide/server-pro/job_launcher/job_launcher.html#configuration.html), and more information about using Workbench Jobs can be found in the [Posit Workbench User Guide](https://docs.posit.co/ide/server-pro/user/rstudio-pro/guide/workbench-jobs.html).
```

File: docs/user/rstudio/ide/guide/environments/r/managing-r.qmd [hunk 2]
```diff
@@ -5,7 +5,7 @@ date-meta: 2022-12-12
...
 
-This section is a practical guide on how to set particular options on R startup. General information on how to manage R package environments is available at [environments.rstudio.com](https://environments.rstudio.com/), and a deeper treatment of R process startup is available in [this article](https://rviews.rstudio.com/2017/04/19/r-for-enterprise-understanding-r-s-startup/) or [What They Forgot to Teach You About R](https://rstats.wtf/r-startup.html).
+This section is a practical guide on how to set particular options on R startup. General information on how to manage R package environments is available at [Reproducible Environments](https://solutions.posit.co/envs-pkgs/environments/), and a deeper treatment of R process startup is available in [this article](https://rviews.rstudio.com/2017/04/19/r-for-enterprise-understanding-r-s-startup/) or [What They Forgot to Teach You About R](https://rstats.wtf/r-startup.html).
```
Comment (jgutman) [inline docs/user/rstudio/ide/guide/environments/r/managing-r.qmd:8]:
```suggestion
This section is a practical guide on how to set particular options on R startup. General information on how to manage R package environments is available at [Reproducible Environments](https://solutions.posit.co/envs-pkgs/environments/), and a deeper treatment of R process startup is available in [What They Forgot to Teach You About R](https://rstats.wtf/r-startup.html).
```

Comment (jgutman) [reply docs/user/rstudio/ide/guide/environments/r/managing-r.qmd:8]: I'm thinking we don't need to be referencing a 6.5 year old blog post here

**Review by jgutman — APPROVED** [review]

**Pull request merged**